lib/wakes/redirect_mapper.rb

```diff
@@ -1,28 +1,28 @@
 # frozen_string_literal: true
 module Wakes
   class RedirectMapper
-    attr_accessor :source, :target, :label
+    attr_accessor :source_host, :source_path, :target_host, :target_path, :label
 
-    def initialize(source, target, label = nil)
-      @source = source
-      @target = target
+    def initialize(source_path, target_path, label = nil, source_host = nil, target_host = nil)
+      @source_host = source_host
+      @source_path = source_path
+      @target_host = target_host
+      @target_path = target_path
       @label = label
-      puts Wakes.color(:white, "Redirecting from #{source} to #{target}", :bold => true)
 
+      puts Wakes.color(:white, "Redirecting from #{source_host}#{source_path} to #{target_host}#{target_path}",
+                       :bold => true)
       print_graph('Starting graph')
       both_present || target_present || source_present || none_present
-      print_graph('Ending graph')
-      puts ''
+      print_graph('Ending graph', :newline => true)
     end
 
     private
 
-    def print_graph(notice)
+    def print_graph(notice, options = {})
       puts Wakes.color(:magenta, notice)
-
-      Wakes::Resource.where(:id => resources_affected).each do |resource|
-        puts resource.to_s
-      end
+      Wakes::Resource.where(:id => resources_affected).each { |resource| puts resource.to_s }
+      puts '' if options[:newline]
```

Comment:
yeah, at some point I need to just refactor this class down into a bunch of smaller classes. Feel free to add in the rubocop comments to ignore class length cop.